__DEFINES/subsystem.dm

```diff
@@ -1,21 +1,57 @@
-// Defines for the subsystem init orders.
+// Subsystem defines.
+// All in one file so it's easier to see what everything is relative to.
 
-#define INIT_RUST               26
-#define INIT_SUPPLY_SHUTTLE     25
-#define INIT_SUN                24
-#define INIT_GARBAGE            23
-#define INIT_JOB                22
-#define INIT_GENETICS           21
-#define INIT_MAP                20
-#define INIT_POWER              19
-#define INIT_OBJECT             18
-#define INIT_XENOARCH           17
-#define INIT_MORE_INIT          16
-#define INIT_AIR                15
-#define INIT_LIGHTING           14
+#define SS_INIT_TICKER_SPAWN       999
+#define SS_INIT_RUST               26
+#define SS_INIT_SUPPLY_SHUTTLE     25
+#define SS_INIT_SUN                24
+#define SS_INIT_GARBAGE            23
+#define SS_INIT_JOB                22
+#define SS_INIT_GENETICS           21
+#define SS_INIT_MAP                20
+#define SS_INIT_POWER              19
+#define SS_INIT_OBJECT             18
+#define SS_INIT_PIPENET            17.5
+#define SS_INIT_XENOARCH           17
+#define SS_INIT_MORE_INIT          16
+#define SS_INIT_AIR                15
+#define SS_INIT_LIGHTING           14
+#define SS_INIT_UNSPECIFIED        0
+#define SS_INIT_EMERGENCY_SHUTTLE -19
+#define SS_INIT_ASSETS            -20
+#define SS_INIT_TICKER            -21
+#define SS_INIT_FINISH            -22
+#define SS_INIT_MINIMAP           -23
 
-#define INIT_EMERGENCY_SHUTTLE -19
-#define INIT_ASSETS            -20
-#define INIT_TICKER            -21
-#define INIT_FINISH            -22
-#define INIT_MINIMAP           -23
+
+#define SS_PRIORITY_TICKER         200
+#define SS_PRIORITY_MOB            150
+#define SS_PRIORITY_NANOUI         120
+#define SS_PRIORITY_VOTE           110
+#define SS_PRIORITY_OBJECTS        100
+#define SS_PRIORITY_POWER          95
+#define SS_PRIORITY_MACHINERY      90
+#define SS_PRIORITY_PIPENET        85
+#define SS_PRIORITY_LIGHTING       80
+#define SS_PRIORITY_AIR            70
+#define SS_PRIORITY_DISEASE        60
+#define SS_PRIORITY_FAST_MACHINERY 55
+#define SS_PRIORITY_EVENT          52
+#define SS_PRIORITY_UNSPECIFIED    50
+#define SS_PRIORITY_SUN            10
+#define SS_PRIORITY_GARBAGE        0
+#define SS_PRIORITY_INACTIVITY    -10
```

Comment:
I don't know that the MC can handle negative/zero priorities properly, their primary use is weighing how the mc splits up a byond tick. if two (or more) subsystems need to run in one tick, it weights how much of the byond tick to give each subsystem based on this number, but its simple math.

I should maybe change the comment on that variable a touch to reflect that better.

Also i'm shocked a 0 priority subsystem isn't flooding you with divide by 0 runtimes. i'm 99% sure it divides by this number.